tests/vmi_lifecycle_test.go

```diff
@@ -422,6 +431,42 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
 					return n.Labels[v1.VirtHandlerHeartbeat]
 				}()).ShouldNot(Equal(timestamp), "Should not have old vmi heartbeat")
 			})
+
+			It("device plugins should re-register if the kubelet restarts", func() {
+
+				By("starting a VMI on a node")
+				vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+				Expect(err).To(BeNil(), "Should submit VMI successfully")
+
+				// Start a VirtualMachineInstance
+				nodeName := tests.WaitForSuccessfulVMIStart(vmi)
+
+				By("triggering a device plugin re-registration on that node")
+				pod, err := kubecli.NewVirtHandlerClient(virtClient).ForNode(nodeName).Pod()
+				Expect(err).ToNot(HaveOccurred())
+
+				_, _, err = tests.ExecuteCommandOnPodV2(virtClient, pod,
+					"virt-handler",
+					[]string{
+						"rm",
+						// We want to fail if the file does not exist, but don't want to be asked
+						// if we really want to remove write-protected files
+						"--interactive=never",
+						device_manager.SocketPath(device_manager.KVMName),
+					})
+				Expect(err).ToNot(HaveOccurred())
+
+				// Give the kubelet and virt-handler some time to re-synchronize
```

Comment:
Is there something we can key off here to prove the re-registration took place? Like maybe a log message or something?

Reply:
Done